### PR TITLE
Add errors to `run_background_receive`

### PR DESCRIPTION
**Pull request by kmchicoine**
Added new errors and error handling for unexpected exits in `run_background_receive` in the web server

### CODE REVIEW

File: src/traits/networking/centralized_web_server_network.rs [hunk 1]
```diff
@@ -285,6 +289,9 @@ impl<
                 vote_index = 0;
             }
         }
+        Err(NetworkError::CentralizedWebServer {
+            source: CentralizedWebServerNetworkError::ClientDisconnected,
+        })
```
Comment (DieracDelta) [inline src/traits/networking/centralized_web_server_network.rs:294]: If this loop stops running, does that mean the Client is disconnected?  Wondering what the Client is in this case and what the difference is between this and the replica being shutdown (e.g. `NetworkError::ShutDown`)

**Review by DieracDelta — APPROVED** [review]
I have one conceptual question, but other than that LGTM!

File: src/traits/networking/centralized_web_server_network.rs (hunk 1 above)
Comment (kmchicoine) [reply src/traits/networking/centralized_web_server_network.rs:294]: Hmm good question- the only way for the polling loop to exit is for the `inner.running` boolean to become false, which happens in the ConnectedNetwork `shutdown()` function, so you're right: `NetworkError::ShutDown` would probably make more sense here

Comment (DieracDelta) [reply src/traits/networking/centralized_web_server_network.rs]: Ah, gotcha! 👍

**Review by DieracDelta — APPROVED** [review]

**Review by elliedavidson — APPROVED** [review]

**Pull request merged**